conda/environment.yml

```
@@ -3,27 +3,164 @@
+  - jupyterhub-singleuser~=4.0.1
+  # nbgitpuller is very helpful when distributing user content
+  - nbgitpuller~=1.2.0
+  # Specify major version of jupyterlab to use. Manually bump it when you want to upgrade
+  # jupyterlab version
+  - jupyterlab>=4.0,<4.1
+  # This provides memory usage stats on JupyterLab and classic notebook
+  - jupyter-resource-usage~=1.0.1
+  # https://syncthing.net/ provides a dropbox-like interface for syncing directories across users,
+  # hubs and machines. Used along with `jupyter-syncthing-proxy`.
+  # - jupyter-syncthing-proxy==1.0.3
+  # - syncthing~=1.22.1
+  # gh-scoped-creds allows users to securely push to GitHub from their repo.
+  # https://blog.jupyter.org/securely-pushing-to-github-from-a-jupyterhub-3ee42dfdc54f has
+  # some more information
+  # - gh-scoped-creds==4.1
+  # More git tools from github-cli
+  - gh~=2.32.1
+
+  # VS Code support
+  # - jupyter-vscode-proxy~=0.5
+  # - code-server~=4.16.1
+
+  # Jupyter tools
+  - ipython~=8.20.0
+  - ipywidgets~=8.1.1
+  - jupyter-ai~=2.20
+  - jupyter-book~=1.0.0
+#  - jupyter-offlinenotebook~=0.2.2 # Outdated according to jupyterlab
+#  - jupyter-collaboration~=2.0.1
+  - jupyterlab-favorites~=3.2.1
+  - jupyterlab-geojson~=3.4.0
+  - jupyterlab-git~=0.50.0
+  - jupyterlab-h5web~=11.1.0
+  - jupyterlab-myst~=2.4.2
+#  - jupyterlab-plotly~=5.18.0 # Outdated accoring to jupyterlab
+  - jupyterlab_pygments~=0.3.0 # To bring extension uptodate
+  - jupytext~=1.16.1
+  - nbdime~=4.0.1
+  # JupyterBook Addons
   - sphinxcontrib-bibtex
-  # For building the Splashpage
-  - cookiecutter
-  # Dependencies of jinja-markdown:
-  - jinja2>=2.11
-  - pygments>=2.6.1
-  - pymdown-extensions>=7.1
-  - pip:
-    - jinja-markdown==1.210911
+  # dashboards
+  - voila~=0.5
+
+  # Core scientific python stack
+  - cython~=3.0.10 # optimization, C API access
+  - flox~=0.9.7 # optimization, xarray operations
+  - hypothesis==6.58.0 # needed by numpy testing tools
+  - networkx==2.8.8
+  - numba~=0.60.0 # high-performance numerics
+  - numpy==1.26.4
+  - pandas>=2.0.1,<3.0
+  - scikit-image~=0.24.0
+  - scikit-learn>=1.2.2,<2.0.0
+  - scipy>=1.10
+  - statsmodels==0.13.5
+  - sympy~=1.12
+  - xarray>=2024.05.0
+
+  # Visualization packages
+  - bokeh~=3.2.2
+  - cartopy~=0.23.0 # geospatial plotting with matplotlib
+  - geemap~=0.32
+  - geoviews~=1.10.1
+  - hvplot~=0.10.0
+  - ipyleaflet~=0.19.2
+  - ipympl~=0.9.3 # This enables matplotlib interaction with jupyter widgets
+  - leafmap~=0.36
+  - lonboard~=0.9
+  - matplotlib~=3.9.0
+  - plotly~=5.18.0
+  - seaborn~=0.13.2 # statistical plotting with matplotlib
+
+  # Geospatial data packages
+  - geopandas>=0.14.4
+  - pygmt~=0.12.0
+  - rasterio~=1.3
+  - rioxarray~=0.17.0
+
+  # File formats and file management, download, dataset caching
+  - h5py~=3.11
+  - h5netcdf~=1.3.0
+  - nco~=5.1.6
+  - pooch~=1.8.0
+  - zarr~=2.18
+  - kerchunk~=0.2
+
+  # Cloud access tools and libraries
+  - awscli~=1.27.76
```

Loosen pin on awscli to get Sphinx version 7 (they both depend on colorama):

```
Could not solve for environment specs
The following packages are incompatible
├─ awscli ~=1.27.76  is installable with the potential options
│  ├─ awscli [1.27.100|1.27.101|...|1.27.99] would require
│  │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  ├─ awscli [1.27.100|1.27.101|...|1.27.99] would require
│  │  └─ colorama >=0.2.5,<0.4.5  with the potential options
│  │     ├─ colorama [0.3.7|0.3.9] would require
│  │     │  └─ python 2.7* , which can be installed;
│  │     ├─ colorama 0.3.7 would require
│  │     │  └─ python 3.4* , which can be installed;
│  │     ├─ colorama [0.3.7|0.3.9] would require
│  │     │  └─ python 3.5* , which can be installed;
│  │     ├─ colorama [0.3.7|0.3.9] would require
│  │     │  └─ python 3.6* , which can be installed;
│  │     └─ colorama [0.3.9|0.4.0|0.4.1|0.4.3|0.4.4], which can be installed;
│  ├─ awscli [1.27.100|1.27.101|...|1.27.99] would require
│  │  └─ pypy3.8 >=7.3.11 , which can be installed;
│  ├─ awscli [1.27.100|1.27.101|...|1.27.99] would require
│  │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  ├─ awscli [1.27.100|1.27.101|...|1.27.99] would require
│  │  └─ pypy3.9 >=7.3.11 , which can be installed;
│  └─ awscli [1.27.100|1.27.101|...|1.27.99] would require
│     └─ python_abi 3.9.* *_cp39, which can be installed;
├─ python ~=3.11.0  is not installable because it requires
│  └─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported;
└─ sphinx >=7  is not installable because there are no viable options
   ├─ sphinx [7.0.0|7.0.1|...|7.3.7] would require
   │  └─ colorama >=0.4.5 , which conflicts with any installable versions previously reported;
   └─ sphinx [7.4.0|7.4.1|...|8.0.2] would require
      └─ colorama >=0.4.6 , which conflicts with any installable versions previously reported.
```

```suggestion
  - awscli~=1.33
```

---

Oh wait, there's another conflict on `docutils`, let me try another combination...

<details>

```
Could not solve for environment specs
The following packages are incompatible
├─ awscli ~=1.33  is installable with the potential options
│  ├─ awscli [1.33.0|1.33.1|...|1.33.9] would require
│  │  └─ docutils >=0.10,<0.17  with the potential options
│  │     ├─ docutils [0.15.2|0.16|...|0.20.1] would require
│  │     │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  │     ├─ docutils [0.15.2|0.16|0.18|0.18.1|0.19] would require
│  │     │  └─ python_abi 3.7.* *_cp37m, which can be installed;
│  │     ├─ docutils [0.15.2|0.16|0.18|0.18.1] would require
│  │     │  └─ python_abi 3.7 *_pypy37_pp73, which can be installed;
│  │     ├─ docutils [0.15.2|0.18.1|0.19] would require
│  │     │  └─ python_abi 3.8 *_pypy38_pp73, which can be installed;
│  │     ├─ docutils [0.15.2|0.16|...|0.20.1] would require
│  │     │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  │     ├─ docutils [0.15.2|0.18.1|0.19] would require
│  │     │  └─ python_abi 3.9 *_pypy39_pp73, which can be installed;
│  │     ├─ docutils [0.15.2|0.16|...|0.20.1] would require
│  │     │  └─ python_abi 3.9.* *_cp39, which can be installed;
│  │     ├─ docutils [0.12|0.13.1|...|0.16] would require
│  │     │  └─ python [2.7* |>=2.7,<2.8.0a0 ], which can be installed;
│  │     ├─ docutils 0.12 would require
│  │     │  └─ python 3.4* , which can be installed;
│  │     ├─ docutils [0.12|0.13.1|0.14] would require
│  │     │  └─ python [3.5* |>=3.5,<3.6.0a0 ], which can be installed;
│  │     ├─ docutils [0.13.1|0.14] would require
│  │     │  └─ python 3.6* , which can be installed;
│  │     ├─ docutils [0.14|0.15|0.15.1|0.15.2|0.16] would require
│  │     │  └─ python >=3.6,<3.7.0a0 , which can be installed;
│  │     ├─ docutils [0.14|0.15|0.15.1|0.15.2|0.16] would require
│  │     │  └─ python >=3.7,<3.8.0a0 , which can be installed;
│  │     ├─ docutils [0.15.2|0.16], which can be installed;
│  │     ├─ docutils [0.15.2|0.16] would require
│  │     │  └─ python_abi 3.6.* *_cp36m, which can be installed;
│  │     ├─ docutils [0.15.2|0.16] would require
│  │     │  └─ python_abi 3.6 *_pypy36_pp73, which can be installed;
│  │     ├─ docutils [0.15.2|0.16] would require
│  │     │  └─ python >=3.8,<3.9.0a0 , which can be installed;
│  │     ├─ docutils 0.16 would require
│  │     │  └─ python_abi 2.7.* *_cp27mu, which can be installed;
│  │     ├─ docutils [0.16|0.20.1] would require
│  │     │  └─ python_abi 3.12.* *_cp312, which can be installed;
│  │     └─ docutils [0.16|0.20.1] would require
│  │        └─ pypy3.9 >=7.3.13 , which can be installed;
│  ├─ awscli [1.33.0|1.33.1|...|1.33.9] would require
│  │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  ├─ awscli [1.33.0|1.33.1|...|1.33.9] would require
│  │  └─ python_abi 3.12.* *_cp312, which can be installed;
│  ├─ awscli [1.33.0|1.33.1|...|1.33.9] would require
│  │  └─ python_abi 3.8.* *_cp38, which can be installed;
│  ├─ awscli [1.33.0|1.33.1|...|1.33.9] would require
│  │  └─ pypy3.9 >=7.3.15 , which can be installed;
│  └─ awscli [1.33.0|1.33.1|...|1.33.9] would require
│     └─ python_abi 3.9.* *_cp39, which can be installed;
├─ jupyter-book ~=1.0.0  is installable and it requires
│  └─ sphinx >=5,<8  with the potential options
│     ├─ sphinx [5.0.0|5.0.1|...|5.3.0], which can be installed;
│     ├─ sphinx [6.0.0|6.1.0|...|7.0.0] would require
│     │  └─ docutils [>=0.18,<0.20 |>=0.18.1,<0.20 ] with the potential options
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|0.18|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|0.18|0.18.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     └─ docutils 0.19 conflicts with any installable versions previously reported;
│     ├─ sphinx [7.0.1|7.1.1|...|7.2.6] would require
│     │  └─ docutils >=0.18.1,<0.21  with the potential options
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|0.18|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|0.18|0.18.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils 0.19 conflicts with any installable versions previously reported;
│     │     ├─ docutils [0.16|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.16|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.20|0.20.1] conflicts with any installable versions previously reported;
│     │     ├─ docutils [0.20|0.20.1] would require
│     │     │  └─ pypy3.8 >=7.3.11 , which can be installed;
│     │     ├─ docutils [0.20|0.20.1] would require
│     │     │  └─ pypy3.9 >=7.3.11 , which can be installed;
│     │     └─ docutils 0.20.1 would require
│     │        └─ pypy3.9 >=7.3.12 , which can be installed;
│     ├─ sphinx [7.3.5|7.3.6|7.3.7] would require
│     │  └─ docutils >=0.18.1,<0.22  with the potential options
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|0.18|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|0.18|0.18.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.18.1|0.19], which can be installed (as previously explained);
│     │     ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils 0.19 conflicts with any installable versions previously reported;
│     │     ├─ docutils [0.16|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.16|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.20|0.20.1] conflicts with any installable versions previously reported;
│     │     ├─ docutils [0.20|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils [0.20|0.20.1], which can be installed (as previously explained);
│     │     ├─ docutils 0.20.1, which can be installed (as previously explained);
│     │     └─ docutils [0.21.1|0.21.2] conflicts with any installable versions previously reported;
│     └─ sphinx [7.4.0|7.4.1|...|7.4.7] would require
│        └─ docutils >=0.20,<0.22  with the potential options
│           ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│           ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│           ├─ docutils [0.15.2|0.16|...|0.20.1], which can be installed (as previously explained);
│           ├─ docutils [0.16|0.20.1], which can be installed (as previously explained);
│           ├─ docutils [0.16|0.20.1], which can be installed (as previously explained);
│           ├─ docutils [0.20|0.20.1] conflicts with any installable versions previously reported;
│           ├─ docutils [0.20|0.20.1], which can be installed (as previously explained);
│           ├─ docutils [0.20|0.20.1], which can be installed (as previously explained);
│           ├─ docutils 0.20.1, which can be installed (as previously explained);
│           └─ docutils [0.21.1|0.21.2] conflicts with any installable versions previously reported;
├─ python ~=3.11.0  is not installable because it conflicts with any installable versions previously reported;
└─ sphinx >=6  is installable with the potential options
   ├─ sphinx [6.0.0|6.1.0|...|7.0.0], which can be installed (as previously explained);
   ├─ sphinx [7.0.1|7.1.1|...|7.2.6], which can be installed (as previously explained);
   ├─ sphinx [7.3.5|7.3.6|7.3.7], which can be installed (as previously explained);
   ├─ sphinx [7.4.0|7.4.1|...|7.4.7], which can be installed (as previously explained);
   └─ sphinx [8.0.0|8.0.2] conflicts with any installable versions previously reported.
```

</details>

---

So awscli v1 has an upper pin on `docutils<0.17`, and sphinx v7.3.0 would require `docutils>=0.20`. We'll need to bump to awscli v2 then (+upgrade ipython, see below):

```suggestion
  - awscli~=2.17
```